python/pyspark/sql/types.py

```diff
@@ -168,10 +168,12 @@ def needConversion(self):
         return True
 
     def toInternal(self, d):
-        return d and d.toordinal() - self.EPOCH_ORDINAL
+        if d is not None:
```

Comment:
Sorry, I see. Should I revert this change? It is mostly for both methods to follow the same logic, as `return x and y` is not a very readable code in my opinion. Also the class `TimestampType` is implemented in a similar way - both methods are starting with input parameter check for None